Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperation.swift

```diff
@@ -599,122 +847,432 @@ struct RenameCollection: TestOperation {
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = (try? container.decode(String.self, forKey: .session)) ?? ""
+        self.collection = try container.decode(String.self, forKey: .collection)
+    }
+
+    func execute(on target: TestOperationTarget, sessionDict: [String: ClientSession]) throws -> TestOperationResult? {
+        guard case let .database(database) = target else {
+            throw TestError(message: "database not provided to dropCollection")
+        }
+        _ = try database.collection(self.collection).drop(session: sessionDict[self.session])
+        return nil
+    }
+}
+
+struct CreateIndex: TestOperation {
+    let session: String
+    let name: String
+    let keys: Document
+
+    private enum CodingKeys: String, CodingKey { case session, name, keys }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = (try? container.decode(String.self, forKey: .session)) ?? ""
+        self.name = try container.decode(String.self, forKey: .name)
+        self.keys = try container.decode(Document.self, forKey: .keys)
+    }
+
+    func execute(on target: TestOperationTarget, sessionDict: [String: ClientSession]) throws -> TestOperationResult? {
+        guard case let .collection(collection) = target else {
+            throw TestError(message: "collection not provided to createIndex")
+        }
+        let indexOptions = IndexOptions(name: self.name)
+        _ = try collection.createIndex(self.keys, indexOptions: indexOptions, session: sessionDict[self.session])
+        return nil
+    }
+}
+
+struct RunCommand: TestOperation {
+    let session: String
+    let command: Document
+    let readPreference: ReadPreference
+
+    private enum CodingKeys: String, CodingKey { case session, command, readPreference }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = (try? container.decode(String.self, forKey: .session)) ?? ""
+        self.command = try container.decode(Document.self, forKey: .command)
+        self.readPreference = (try? container.decode(ReadPreference.self, forKey: .readPreference)) ??
+            ReadPreference.primary
+    }
+
+    func execute(on target: TestOperationTarget, sessionDict: [String: ClientSession]) throws -> TestOperationResult? {
+        guard case let .database(database) = target else {
+            throw TestError(message: "database not provided to runCommand")
+        }
+        let runCommandOptions = RunCommandOptions(readPreference: self.readPreference)
+        let result = try database.runCommand(
+            self.command,
+            options: runCommandOptions,
+            session: sessionDict[self.session]
+        )
+
+        var refinedResult = Document()
+        try result.copyElements(
```

Comment:
instead of stripping out these extra fields by name, we can conform `TestOperationResult` to [`Matchable`](https://github.com/mongodb/mongo-swift-driver/blob/master/Tests/MongoSwiftSyncTests/SpecTestRunner/Match.swift), and update the assertion in the test runner to assert that the result matches the expected one instead of equaling it.

Reply:
Done.